src/main/java/omero/gateway/Gateway.java

```diff
@@ -1072,7 +1073,11 @@ private SessionWrapper createSession(LoginCredentials c)
 
         if (c.isCheckNetwork()) {
             try {
-                String ip = InetAddress.getByName(c.getServer().getHostname())
+                String hn = c.getServer().getHostname();
+                // this could be a websocket URL
+                if (hn.contains("://"))
+                    hn = new URI(hn).getHost();
```

Comment:
Interesting extension of the server specification.

This makes me wonder whether we should always construct an `URI` object and use [URI.isAbsolute()](https://docs.oracle.com/javase/7/docs/api/java/net/URI.html#isAbsolute())
to detect whether there is a scheme and to extract the host name.

Reply:
That's true, using `isAbsolute()` is better then 'manually' looking for a `://`. I'll change that.